script.js

```diff
@@ -1,4 +1,542 @@
+      losingPlayer = player2;
+      losingScore = player2Score;
+    } else if (player2Score > player1Score) {
+      leadingPlayer = player2;
+      leadingScore = player2Score;
+      losingPlayer = player1;
+      losingScore = player1Score;
+    }
+  }
+  if (gameMode == 2 || gameMode == 4 || gameMode == 6) {
+    if (player1Score >= player2Score) {
+      leadingPlayer = player2;
+      leadingScore = player2Score;
+      losingPlayer = player1;
+      losingScore = player1Score;
+    } else if (player2Score > player1Score) {
+      leadingPlayer = player1;
+      leadingScore = player1Score;
+      losingPlayer = player2;
+      losingScore = player2Score;
+    }
+  }
+  return `<br>---Leaderboard---<br>${leadingPlayer} is leading. Score is ${leadingScore}<br>${losingPlayer} is losing. Score is ${losingScore}`;
+};
+
+// function to reset game
+var gameReset = function () {
+  // reset game mode
+  gameMode = 0;
+  // reset game stage
+  gameStage = 1;
+  // switch player
+  currentPlayer = player1;
+  // reset scores
+  player1Score = 0;
+  player2Score = 0;
+  return `Game reset, please choose game mode to start<br>${gameInstruction1}`;
+};
+// game instruction message
+var gameInstruction1 = `Enter '1' for ${standard}<br>Enter '2' for ${reverse} Game<br>Enter '3' for ${auto} Game<br>Enter '4' for ${autoReverse} Game<br>Enter '5' for ${variableDiceStandard} Game<br>Enter '6' for ${variableDiceReverse} Game<br>Enter '7' for ${variablePlayerStandard} Game<br>Enter 'x' to reset game mode`;
+
+var gameInstruction2 = `Please enter the number '1' or '2'.`;
+
+var gameInstruction3 = `Please enter the number(>0) of dice`;
+
+var gameInstruction4 = `Please enter the number(>1) of players`;
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  var myOutputValue = ``;
+  if (input == `x`) {
+    return gameReset();
+  } else if (gameMode == 0) {
+    // check if input is a number
+    if (Number.isNaN(Number(input)) == true) {
+      return gameInstruction1;
+    }
+    // check if input is the number 1 or 2 or 3 or 4 or 5 or 6.
+    else if (
+      input != 1 &&
+      input != 2 &&
+      input != 3 &&
+      input != 4 &&
+      input != 5 &&
+      input != 6 &&
+      input != 7 &&
+      input != 8
+    ) {
+      return gameInstruction1;
+    } else if (input == 1 || input == 2 || input == 3 || input == 4) {
+      // set gamemode
+      gameMode = Number(input);
+      console.log(gameMode);
+      return `${checkGameSelected()} is selected, click submit to start playing`;
+    } else if (input == 5 || input == 6) {
+      // set gamemode
+      gameMode = Number(input);
+      console.log(gameMode);
+      return `${checkGameSelected()} is selected, enter number of dice to start playing`;
+    } else if (input == 7 || input == 8) {
+      // set gamemode
+      gameMode = Number(input);
+      console.log(gameMode);
+      return `${checkGameSelected()} is selected, enter number of player to start playing`;
+    }
```

Comment:
Is there a functional reason why you needed to separate out the inputs like this since the code is roughly the same? You could pass in the gameMode as a parameter to checkGameSelected() function and evaluate that in the helperfunction to decide what gamemode it is. That way, you don't even need this if else block